src/genotyper.cc

```diff
@@ -667,6 +679,26 @@ Status genotype_site(const genotyper_config& cfg, MetadataCache& cache, BCFData&
                                   sample_mapping, variant_records, adh, min_ref_depth,
                                   genotypes, loss_trackers));
         }
+
+        if (residuals != nullptr) {
+            bool any_lost_calls = false;
+            for (int i = 0; i < bcf_nsamples; i++) {
+                if (genotypes[sample_mapping.at(i)*2].RNC != NoCallReason::N_A ||
+                    genotypes[sample_mapping.at(i)*2 + 1].RNC != NoCallReason::N_A) {
+                    any_lost_calls = true;
+                    break;
+                }
+            }
+            if (!any_lost_calls)
+                continue;
```

Comment:
This `continue` may bite us later if we add more code after this block. Let's flip the `if` and let the loop body play itself out.